oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/invalidData/InvalidIndexDefinitionTest.java

```diff
@@ -0,0 +1,259 @@
+                .with((Observer) provider)
+                .with(editorProvider)
+                .with(new PropertyIndexEditorProvider())
+                .with(new NodeTypeIndexProvider())
+                .createContentRepository();
+    }
+    
+    @Test
+    public void allFine() throws CommitFailedException {
+        createIndexNodeAndData();
+        root.commit();
+        String query = "select [jcr:path] from [nt:base] where isdescendantnode('/tmp') and upper([test]) = 'HELLO'";
+        assertThat(explain(query), containsString("lucene:test"));
+        assertQuery(query, Lists.newArrayList("/tmp/testNode"));
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void invalidCodec() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // An incorrect value throws an exception, for example:
+        // java.lang.IllegalArgumentException: A SPI class of type org.apache.lucene.codecs.Codec with name 'Lucene46x' does not exist.
+        String codecValue = Codec.getDefault().getName() + "x";
+        def.setProperty(LuceneIndexConstants.CODEC_NAME, codecValue);
+        root.commit();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void invalidCompatMode() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // 3 results in IllegalArgumentException: Unknown version : 3
+        def.setProperty(LuceneIndexConstants.COMPAT_MODE, 3);
+        root.commit();
+    }
+    
+    @Test(expected = PatternSyntaxException.class)
+    public void invalidValueRegex() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // An incorrect value, for example "[a-z", results in
+        // java.util.regex.PatternSyntaxException: Unclosed character class near index 3
+        def.setProperty(LuceneIndexConstants.PROP_VALUE_REGEX, "[a-z");
+        root.commit();
+    }
+    
+    @Test(expected = PatternSyntaxException.class)
+    public void invalidQueryFilterRegex() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // An incorrect value, for example "[a-z", results in
+        // java.util.regex.PatternSyntaxException: Unclosed character class near index 3
+        def.setProperty(LuceneIndexConstants.PROP_QUERY_FILTER_REGEX, "[a-z");
+        root.commit();
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void invalidBlobSize() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // 1L + Integer.MAX_VALUE results in IllegalArgumentException: Out of range: 2147483648
+        def.setProperty("blobSize", 1L + Integer.MAX_VALUE);
+        root.commit();
+    }    
+
+    @Test
+    public void negativeBlobSize() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // 1L + Integer.MAX_VALUE results in IllegalArgumentException: Out of range: 2147483648
```

Comment:
Comment can be fixed here to point out the expected behaviour when this value is set to a negative number.